creme/metrics/_confusion_matrix.pyx

```diff
@@ -0,0 +1,232 @@
+import warnings
+
+import numpy as np
+cimport numpy as np
+
+from collections import defaultdict
+
+cdef bint boolean_variable = True
+
+cdef class ConfusionMatrix:
+    """ Confusion Matrix for binary-class and multi-class classification.
+
+    Parameters
+    ----------
+    classes: set, list (default=None)
+        The initial set of classes.
+
+    Notes
+    -----
+    This confusion matrix is a 2D matrix of shape ``(n_classes, n_classes)``, corresponding
+    to a single-target (binary and multi-class) classification task.
+
+    Each row represents ``true`` (actual) class-labels, while each column corresponds
+    to the ``predicted`` class-labels. For example, an entry in position ``[1, 2]`` means
+    that the true class-label is 1, and the predicted class-label is 2 (incorrect prediction).
+
+    This structure is used to keep updated statistics about a single-output classifier's
+    performance and to compute multiple evaluation metrics.
+
+    """
+    _fmt = '0.0f'
+
+    def __init__(self, classes=None):
+        self._init_classes = set(classes) if classes is not None else set()
+        self.classes = self._init_classes
+        self.sum_diag = 0.0
+        self.sum_row = defaultdict(float)
+        self.sum_col = defaultdict(float)
+        self.data = defaultdict(lambda: defaultdict(float))
+
+    def __setitem__(self, entry, double sample_weight):
+        if len(entry) != 2:
+            raise KeyError(f'Expected (true_idx, pred_idx) entry, received: {entry}')
+        y_true, y_pred = entry
+        self.update(y_true, y_pred, sample_weight)
+
+    def __getitem__(self, key):
+        """Syntactic sugar for accessing the counts directly."""
+        return self.data[key]
+
+    def update(self, y_true, y_pred, sample_weight=1.):
+        if sample_weight is None:
+            # Since we ca not set a default value in the signature
```

Comment:
cannot

Reply:
Fixed typo in a61a9597c360687d911e4a90eaded8f8088b06cd